File: articles/institutional-incentives-behavioral-responses/go/institutional_incentives_score.go
package main

import "fmt"

type InstitutionalIncentiveCase struct {
	ValueAlignment     float64
	Fairness           float64
	InformationQuality float64
	Legitimacy         float64
	LearningSupport    float64
	Accountability     float64
	BiasPressure       float64
	MetricSubstitution float64
	ReportingDistortion float64
	BehavioralBurden   float64
	ShortTermism       float64
	StatusInequality   float64
	MotivationCrowding float64
}

func InstitutionalIncentiveScoreRaw(x InstitutionalIncentiveCase) float64 {
	return 0.14*x.ValueAlignment +
		0.12*x.Fairness +
		0.13*x.InformationQuality +
		0.12*x.Legitimacy +
		0.12*x.LearningSupport +
		0.10*x.Accountability -
		0.10*x.BiasPressure -
		0.12*x.MetricSubstitution -
		0.09*x.ReportingDistortion -
		0.08*x.BehavioralBurden -
		0.07*x.ShortTermism -
		0.06*x.StatusInequality -
		0.05*x.MotivationCrowding
}

func main() {
	demo := InstitutionalIncentiveCase{82, 76, 80, 78, 75, 72, 25, 22, 20, 28, 24, 18, 21}
	fmt.Printf("Institutional incentive raw score: %.2f\n", InstitutionalIncentiveScoreRaw(demo))
}

File: articles/institutional-incentives-behavioral-responses/fortran/institutional_incentives_score.f90
program institutional_incentives_score
  implicit none

  real :: value_alignment, fairness, information_quality, legitimacy
  real :: learning_support, accountability, bias_pressure, metric_substitution
  real :: reporting_distortion, behavioral_burden, short_termism
  real :: status_inequality, motivation_crowding, score

  value_alignment = 82.0
  fairness = 76.0
  information_quality = 80.0
  legitimacy = 78.0
  learning_support = 75.0
  accountability = 72.0
  bias_pressure = 25.0
  metric_substitution = 22.0
  reporting_distortion = 20.0
  behavioral_burden = 28.0
  short_termism = 24.0
  status_inequality = 18.0
  motivation_crowding = 21.0

  score = 0.14 * value_alignment + &
          0.12 * fairness + &
          0.13 * information_quality + &
          0.12 * legitimacy + &
          0.12 * learning_support + &
          0.10 * accountability - &
          0.10 * bias_pressure - &
          0.12 * metric_substitution - &
          0.09 * reporting_distortion - &
          0.08 * behavioral_burden - &
          0.07 * short_termism - &
          0.06 * status_inequality - &
          0.05 * motivation_crowding

  print *, "Institutional incentive raw score:", score
end program institutional_incentives_score
